pkg/processor/build/builder.go

```diff
@@ -657,7 +697,7 @@ func (b *Builder) createTempFileFromYAML(fileName string, unmarshalledYAMLConten
 	tempFileName := path.Join(os.TempDir(), fileName)
 
 	// write the temporary file
-	ioutil.WriteFile(tempFileName, marshalledFileContents, os.FileMode(0644))
+	ioutil.WriteFile(tempFileName, marshalledFileContents, os.FileMode(0744))
```

Comment:
Why do we need execute for this and any of the other directories?

Reply:
The other directories need `+x` permission for docker to run properly, by what I saw. Was pretty sure I noticed and reverted this line, but apparently not

Reply:
Can you figure out why and document it?